include/gc/Transforms/CMakeLists.txt

```diff
@@ -2,7 +2,7 @@ if(GC_ENABLE_DNNL)
     list(APPEND TABLEGEN_MACROS -DGC_HAS_ONEDNN_DIALECT)
 endif()
 if(GC_ENABLE_IMEX)
-    list(APPEND TABLEGEN_MACROS -DGC_USE_IMEX)
+    list(APPEND TABLEGEN_MACROS -DGC_ENABLE_IMEX)
```

Comment:
After #200 there were mixed usage of `GC_USE_IMEX/GC_ENABLE_IMEX/GC_UNABLE_GPU` that caused `imex` build to fail. In my PR I fix all the variable names to be `GC_ENABLE_IMEX`

Reply:
There's a distinction between `GC_ENABLE_IMEX` and the `GC_USE_IMEX`: the former is a cmake variable, the latter - a macro used in the code. `GC_ENABLE_GPU` should go.